src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java

```diff
@@ -275,18 +279,52 @@ private AnomalyDetectorSettings() {}
 
     public static final int MULTI_ENTITY_NUM_TREES = 10;
 
-    // cache related
-    public static final int DEDICATED_CACHE_SIZE = 10;
+    // ======================================
+    // cache related parameters
+    // ======================================
+    // Each detector has its dedicated cache that stores ten entities' states per node. A detector's hottest entities load their states into
+    // the dedicated cache. Other detectors cannot use space reserved by a detector's dedicated cache. Previously, the size of the dedicated
+    // cache is not dynamically adjustable by users. This PR creates a dynamic setting AnomalyDetectorSettings.DEDICATED_CACHE_SIZE to make
+    // dedicated cache's size flexible. When that setting is changed, if the size decreases, we will release memory if required (e.g., when
+    // a user also decreased AnomalyDetectorSettings.MODEL_MAX_SIZE_PERCENTAGE, the max memory percentage that AD can use); if the size
+    // increases, we may reject the setting change if we cannot fulfill that request (e.g., when it will uses more memory than allowed for
+    // AD).
+
+    /*
+     * Opensearch-only setting
+     * Each detector has its dedicated cache that stores ten entities' states per node.
+     * A detector's hottest entities load their states into the dedicated cache.
+     * Other detectors cannot use space reserved by a detector's dedicated cache.
+     * DEDICATED_CACHE_SIZE is a setting to make dedicated cache's size flexible.
+     * When that setting is changed, if the size decreases, we will release memory
+     * if required (e.g., when a user also decreased AnomalyDetectorSettings.MODEL_MAX_SIZE_PERCENTAGE,
+     * the max memory percentage that AD can use);
+     * if the size increases, we may reject the setting change if we cannot fulfill
+     * that request (e.g., when it will uses more memory than allowed for AD).
+     *
+     * With compact rcf, rcf with 30 trees and shingle size 4 is of 500KB.
+     * The recommended max heap size is 32 GB. Even if users use all of the heap
+     * for AD, the max number of entity model cannot surpass 3.2 * 10^10 / 5*10^5 = 6.4 * 10 ^4
```

Comment:
Takes some time to understand these numbers in `3.2 * 10^10 / 5*10^5`. I think you mean 3.2 GB/500KB  and 3.2 GB is from 10% memory limit of AD plugin?

Reply:
yes, added.